Assets/Packages/com.studio23.ss2.upmassistant/Editor/Manager/PackageHandler.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Studio23.SS2.UPMAssistant.Editor.Data;
+using UnityEditor;
+using UnityEngine;
+
+namespace Studio23.SS2.UPMAssistant.Editor
+{
+    
+    public static class PackageHandler
+    {
+        public static string FilePath;
+        public static PackageJsonData LoadData()
+        {
+
+
+            if (File.Exists(FilePath))
+            {
+                var jsonString = File.ReadAllText(FilePath);
+                
+                if (string.IsNullOrWhiteSpace(jsonString))
+                {
+                    Debug.Log($"package.json is empty. Default data is loaded");
+                     return LoadDefaultPackageJsonData();
+                }
+                Debug.Log($"Previous data is loaded");
+               return  JsonConvert.DeserializeObject<PackageJsonData>(jsonString);
+            }
+
+            Debug.Log($"Default data loaded");
+            return  LoadDefaultPackageJsonData();
+        }
+
+        public static PackageJsonData CheckUpdateOnDisabledProperties(PackageJsonData jsonData)
+        {
+            var haveUpdate = UpdateProperties(jsonData);
+            if (haveUpdate)
+            {
+                SaveData(jsonData);
+            }
+
+            return jsonData;
+        }
+
+        public static void SaveData(PackageJsonData newJsonData)
+        {
+            string jsonString = JsonConvert.SerializeObject(newJsonData, Formatting.Indented);
+            JObject jsonObject = JObject.Parse(jsonString);
```

Comment:
What is the reason behind using JObject?

Reply:
try to parse json object to check that has json data.
we can check string is empty or not too